src/test/kotlin/com/exactpro/th2/codec/moldudp64/TestMoldUdp64Codec.kt

```diff
@@ -0,0 +1,529 @@
+            sequence = th2Sequence
+
+            metadataBuilder.apply {
+                timestamp = Instant.now().toTimestamp()
+                putAllProperties(th2MetadataProperties)
+            }
+
+            addField(SESSION_FIELD, packetSession)
+            addField(SEQUENCE_FIELD, packetSequenceNumber)
+            addField(COUNT_FIELD, packetMessageCount)
+            addField(LENGTHS_FIELD, packetMessageLengths)
+        }
+
+        private fun message(messageBody: ByteArray): RawMessage.Builder = RawMessage.newBuilder().setBody(ByteString.copyFrom(messageBody))
+
+        fun parsed(
+            header: Message.Builder,
+            vararg messages: RawMessage.Builder
+        ): MessageGroup = MessageGroup.newBuilder().run {
+            this += header
+
+            messages.forEachIndexed { index, message ->
+                this += message.apply {
+                    metadataBuilder.apply {
+                        putAllProperties(header.metadataBuilder.propertiesMap)
+                        idBuilder.apply {
+                            connectionIdBuilder.sessionAlias = header.sessionAlias
+                            direction = header.direction
+                            sequence = header.sequence
+                            timestamp = header.metadata.timestamp
+                            addSubsequence(index + 1)
+                        }
+                    }
+                }
+            }
+
+            build()
+        }
+
+        private fun MessageGroup.assertHeaderPresence() {
+            assertTrue(messagesList.isNotEmpty(), "Message group is empty")
+            assertTrue(messagesList[0].hasMessage(), "Header message is not a parsed one")
+            assertEquals(MESSAGE_TYPE, messagesList[0].message.messageType, "Unexpected header message type")
+        }
+
+        private fun MessageGroup.withHeader(block: Message.() -> Unit) {
+            assertHeaderPresence()
+            messagesList[0].message.block()
+        }
+
+        private fun Message.assertSession(session: String) {
+            assertEquals(session, getString(SESSION_FIELD), "Unexpected packet session")
+        }
+
+        private fun Message.assertSequenceNumber(sequenceNumber: ULong) {
+            val parsedSequenceNumber = assertNotNull(getString(SEQUENCE_FIELD), "No sequence number field")
+            assertEquals(sequenceNumber, parsedSequenceNumber.toULong(), "Unexpected packet sequence number")
+        }
+
+        private fun Message.assertMessageCount(messageCount: UShort) {
+            val parsedMessageCount = assertNotNull(getString(COUNT_FIELD), "No message count field")
+            assertEquals(messageCount, parsedMessageCount.toUShort(), "Unexpected packet message count")
+        }
+
+        private fun Message.assertMessageLengths(messageLengths: List<UShort>) {
+            val packetMessageLengths = assertNotNull(getList(LENGTHS_FIELD), "No message lengths field").map {
```

Comment:
Does it mean that the header will have a field with a list of message lengths? We won't decode each length as a separate parsed message (like in SF)?

I am asking just to clarify if I get everything correctly. Don't think that we should repeat the SF here

Reply:
> Does it mean that the header will have a field with a list of message lengths?

Yep